ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/CIOEngine.kt

```diff
@@ -24,11 +26,29 @@ internal class CIOEngine(override val config: CIOEngineConfig) : HttpClientEngin
 
     private val endpoints = ConcurrentHashMap<String, Endpoint>()
 
-    @OptIn(InternalCoroutinesApi::class)
-    private val selectorManager by lazy { ActorSelectorManager(dispatcher) }
+    private val selectorManager = ActorSelectorManager(dispatcher)
 
     private val connectionFactory = ConnectionFactory(selectorManager, config.maxConnectionsCount)
 
+    private val requestsJob: CoroutineContext
+
+    override val coroutineContext: CoroutineContext
+
+    init {
+        val parent = super.coroutineContext[Job]!!
+        requestsJob = SilentSupervisor(parent)
+        coroutineContext = super.coroutineContext + requestsJob
+
+        GlobalScope.launch(super.coroutineContext) {
```

Comment:
❗️ This will cause unclosed selector manager when the context is already cancelled. Use atomic launch mode to prevent it